backend/src/test_fixtures.rs

```diff
@@ -35,36 +45,83 @@ impl NeighbourhoodFixture {
     }
 
     pub fn neighbourhood_map(&self) -> Result<(Neighbourhood, MapModel)> {
-        let (neighborhood_stats, map) = self.neighbourhood_params()?;
+        let (boundary, map) = self.neighbourhood_params()?;
 
         // Uncomment if you want to re-save the savefiles
         // std::fs::write(self.savefile_path(), map.to_savefile().to_string())?;
 
         let edit_perimeter_roads = false;
-        let neighbourhood = Neighbourhood::new(&map, neighborhood_stats, edit_perimeter_roads)?;
+        let neighbourhood = Neighbourhood::new(&map, boundary, edit_perimeter_roads)?;
         Ok((neighbourhood, map))
     }
 
+    pub fn pbf_path(&self) -> String {
+        if self.is_cnt {
+            format!("../web/public/cnt_osm/{}.osm.pbf", self.study_area_name)
+        } else {
+            format!("../web/public/severance_pbfs/{}.pbf", self.study_area_name)
+        }
+    }
+
+    pub fn boundary_path(&self) -> String {
+        if self.is_cnt {
+            format!(
+                "../web/public/cnt_boundaries/{}.geojson",
+                self.study_area_name
+            )
+        } else {
+            format!("../web/public/boundaries/{}.geojson", self.study_area_name)
+        }
+    }
+
+    pub fn population_zones_path(&self) -> Option<String> {
+        if self.is_cnt {
+            Some(format!(
+                "../web/public/cnt_prioritization/population_{}.bin",
+                self.study_area_name
+            ))
+        } else {
+            None
+        }
+    }
+
+    pub fn population_zones(&self) -> Option<Vec<PopulationZone>> {
+        let path = self.population_zones_path()?;
+        let population_zone_bytes =
+            std::fs::read(&path).expect(&format!("unable to read population_zones: {path}"));
+        Some(
+            bincode::deserialize(&population_zone_bytes)
+                .expect("unable to deserialize population_zones"),
+        )
+    }
+
     pub fn map_model_builder(&self) -> Result<impl Fn() -> Result<MapModel> + use<'_>> {
         let study_area_name = &self.study_area_name;
 
-        let pbf_path = format!("../web/public/severance_pbfs/{study_area_name}.pbf");
-        let input_bytes = std::fs::read(&pbf_path)?;
-
-        let boundary_path = format!("../web/public/boundaries/{study_area_name}.geojson");
-        let boundary: Feature = std::fs::read_to_string(&boundary_path)?.parse()?;
-        // All test study area boundaries are polygons for now
-        let polygon: Polygon = boundary.try_into()?;
-        let multi_polygon: MultiPolygon = polygon.into();
+        let input_bytes = std::fs::read(&self.pbf_path())
+            .context(format!("unable to read '{}'", self.pbf_path()))?;
```

Comment:
I've long known about `context` in theory, but I finally used it today! Wow!

Expect more of this in the future! 

In case you aren't familiar... I find rusts `?` easy to write, but it can make tracking down where the error originated tricky. Stack traces are great for this, but only happen when you have a crash, not a `?` propogation.

Context will add this text to the error that gets propagated, so when you fail or crash at an outer level, you'll still see this context string.

Reply:
Of course now that I think about it, this particular code is all test code, so maybe we should just be `unwrap`-ing everywhere anyway... but I think you get the idea.

Reply:
I also haven't used it too much, but agree it's super helpful! For the particular case of getting filenames into errors, I've also used https://docs.rs/fs-err/latest/fs_err/index.html before